packages/eightballer/connections/dcxt/interfaces/asset_bridging.py

```diff
@@ -0,0 +1,93 @@
+import requests
+
+from derive_client.data_types import ChainID, Currency
+from derive_client.clients.async_client import AsyncClient
+
+from packages.eightballer.connections.dcxt import dcxt
+from packages.eightballer.connections.dcxt.dcxt.derive import DeriveClient
+from packages.zarathustra.protocols.asset_bridging.custom_types import BridgeStatus
+from packages.zarathustra.protocols.asset_bridging.message import AssetBridgingMessage
+from packages.zarathustra.protocols.asset_bridging.dialogues import AssetBridgingDialogue, BaseAssetBridgingDialogue
+from packages.eightballer.connections.dcxt.interfaces.interface_base import BaseInterface
+
+
+class AssetBridgingInterface(BaseInterface):
+    """Interface for the asset bridging protocol."""
+
+    protocol_id = AssetBridgingMessage.protocol_id
+    dialogue_class = AssetBridgingDialogue
+    dialogues_class = BaseAssetBridgingDialogues
+
+    async def request_bridge(
+            self, message: AssetBridgingMessage, dialogue: AssetBridgingDialogue, connection
+        ) -> AssetBridgingMessage | None:
+
+        if not message.performative == AssetBridgingMessage.Performative.REQUEST_BRIDGE:
+            raise RuntimeError()
+
+        source_chain: str = message.source_chain
+        target_chain: str = message.target_chain
+        source_token: str = message.source_token
+        target_token: str | None = message.target_token
+        bridge: str = message.bridge
+        kwargs: dict[str, str] | None = message.kwargs
+
+        if not bridge == "derive":
+            raise NotImplementedError()
+        if not (source_chain == "derive" or target_chain == "derive"):
+            raise NotImplementedError()
+        if target_token and target_token != source_token:
+            raise NotImplementedError()
+        
+        ledger_id = exchange_id = bridge
+        exchange: DeriveClient = connection.exchanges[ledger_id][exchange_id]
+        client: AsyncClient = exchange.client
```

Comment:
This is an async client, but the methods on the Derive `BaseClient` (`deposit_to_derive` & `withdraw_from_derive`) are sync method